frontend/institution/allInstitutionsController.js

```diff
@@ -0,0 +1,107 @@
+        $state, AuthService, InstitutionService, MessageService, ngClipboard){
+        var institutionCardCtrl = this;
+        var URL_INSTITUTION = '/institution/';
+
+        institutionCardCtrl.user = AuthService.getCurrentUser();
+
+        institutionCardCtrl.goToInstitution = function goToInstitution(institutionKey) {
+            $state.go('app.institution.timeline', {institutionKey: institutionKey});
+        };
+
+        institutionCardCtrl.generateLink = function generateLink(){
+            var currentUrl = (window.location.host);
+            var url = currentUrl + URL_INSTITUTION + institutionCardCtrl.institution.key + "/home";
+            ngClipboard.toClipboard(url);
+            MessageService.showToast("O link foi copiado");
+        };
+
+        institutionCardCtrl.showFollowButton = function showFollowButton() {
+            return institutionCardCtrl.institution && !institutionCardCtrl.isMember && 
+                   institutionCardCtrl.institution.name !== "Ministério da Saúde" &&
+                   institutionCardCtrl.institution.name !== "Departamento do Complexo Industrial e Inovação em Saúde";
+         };
+
+        institutionCardCtrl.follow = function follow(){
+            var promise = InstitutionService.follow(institutionCardCtrl.institution.key);
+            promise.then(function success(){
+                MessageService.showToast("Seguindo "+ institutionCardCtrl.institution.name);
+                var institution =  {
+                    acronym: institutionCardCtrl.institution.acronym,
+                    key: institutionCardCtrl.institution.key,
+                    photo_url: institutionCardCtrl.institution.photo_url,
+                    legal_nature: institutionCardCtrl.institution.legal_nature,
+                    actuation_area: institutionCardCtrl.institution.actuation_area
+                };
+                institutionCardCtrl.user.follow(institution);
+                AuthService.save();
+            }, function error() {
+                MessageService.showToast('Erro ao seguir a instituição.');
+            });
+            return promise;
+        };
+
+        institutionCardCtrl.unfollow = function unfollow() {
+            if(institutionCardCtrl.user.isMember(institutionCardCtrl.institution.key)){
+                MessageService.showToast("Você não pode deixar de seguir " + institutionCardCtrl.institution.name);
+            }
+            else{
+                var promise = InstitutionService.unfollow(institutionCardCtrl.institution.key);
+                promise.then(function success(){
+                    MessageService.showToast("Deixou de seguir "+institutionCardCtrl.institution.name);
+                    institutionCardCtrl.user.unfollow(institutionCardCtrl.institution);
+                    AuthService.save();
+                }, function error() {
+                    MessageService.showToast('Erro ao deixar de seguir instituição.');
```

Comment:
Let's try to be more friendly, avoiding error messages.

Suggestion:
"Não foi possível seguir a instituição."